### PR TITLE
feat: update runtime v2 doc

### PR DESCRIPTION
**Pull request by caohuilin**
## Summary
Update Docs:

- add `createRoot`、 `render`、`renderString`、`renderStreaming` 、`createRequestHandler` function doc
- mark `bootstrap` and `createApp` will be deprecated
- support `src/entry.tsx` custom entry when enable `source.enableCustomEntry`
- support `src/entry.server.tsx` to custom ssr entry

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/document/main-doc/docs/zh/guides/topic-detail/framework-plugin/hook-list.mdx [hunk 1]
```diff
@@ -467,60 +457,23 @@ import type { CliPlugin, AppTools } from '@modern-js/app-tools';
...
+        });
+        return { entrypoints: newEntryPoints };
       },
     };
   },
 });
 ```
 
-### `modifyEntryRenderFunction`
+### `generateEntryCode`
```
Comment (zllkjc) [inline packages/document/main-doc/docs/zh/guides/topic-detail/framework-plugin/hook-list.mdx:471]: I think the hooks of the entry should not be exposed now

File: packages/document/main-doc/docs/zh/guides/concept/entries.mdx [hunk 2]
```diff
@@ -153,55 +194,40 @@ export default (App: React.ComponentType, bootstrap: () => void) => {
 
 此时，Modern.js 生成的文件内容如下：
```
Comment (zllkjc) [inline packages/document/main-doc/docs/zh/guides/concept/entries.mdx:195]: cloud we remove this part of the code ?

**Comment by zllkjc** [comment]
maybe we can add some summary for the pr

**Comment by caohuilin** [comment]
> maybe we can add some summary for the pr

done

**Review by zllkjc — APPROVED** [review]

**Pull request merged**